ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/SubmissionValidator.scala

```diff
@@ -215,23 +223,55 @@ class SubmissionValidator[LogResult](
               logger.error("Unexpected failure during submission validation.", exception)
               Success(Left(ValidationError(exception.getLocalizedMessage)))
           }
-        }
       case _ =>
         Future.successful(
           Left(ValidationError(s"Failed to parse submission, correlationId=$correlationId")))
     }
 
   private def flattenInputStates(
-      inputs: Map[DamlStateKey, Option[DamlStateValue]]): Map[DamlStateKey, DamlStateValue] =
+      inputs: Map[DamlStateKey, Option[DamlStateValue]]
+  ): Map[DamlStateKey, DamlStateValue] =
     inputs.collect {
       case (key, Some(value)) => key -> value
     }
 
-  object Metrics {
+  private final class TimedLedgerStateAccess(delegate: LedgerStateAccess[LogResult])
+      extends LedgerStateAccess[LogResult] {
+    override def inTransaction[T](
+        body: LedgerStateOperations[LogResult] => Future[T]
+    ): Future[T] = {
+      val acquireStopped = new AtomicBoolean(false)
```

Comment:
@SamirTalwar-DA I guess that if the failure case stops the time, then the inTransaction one won't anymore; on the other hand the inTransaction body might stop it, the transaction might still fail later and then you need to make sure it's not already stopped in the failure halding code (in order not to record a false later time). I think this is the reason why you need to atomic flag but regardless pls. document it in a comment.

Reply:
Yes, good call. I'll also make it a separate timer.

Reply:
Even better indeed